test/mpi.py

```diff
@@ -23,36 +23,251 @@
+        for scriptname in scriptnames:
+            # if the scriptname is an executable located on this machine
+            if mpim.which(scriptname):
+                (returned_scriptname, mpi, found) = mpim.what_mpi(scriptname)
+                print("what mpi returns: %s, %s, %s" % (returned_scriptname, mpi, found))
+                # if an mpi implementation was found
+                if mpi:
+                    self.assertTrue(mpi in found,
+                                    msg="returned mpi (%s) is not an element of found_mpi (%s)" % (mpi, found))
+                    self.assertTrue(returned_scriptname == scriptname,
+                                    msg="returned scriptname (%s) doesn't match actual scriptname (%s)" %
+                                    (returned_scriptname, scriptname))
+                else:
+                    self.assertTrue(returned_scriptname.endswith("mpirun"),
+                                    msg="no mpi found, scriptname should be the path to mpirun but is %s" %
+                                    returned_scriptname)
+
+    def test_stripfake(self):
+        """Test if stripfake actually removes the /bin/fake path in $PATH"""
+        print("old path: %s" % os.environ["PATH"])
+        mpim.stripfake()
+        newpath = os.environ["PATH"]
+        self.assertFalse(("bin/%s" % mpim.FAKE_SUBDIRECTORY_NAME) in newpath, msg="the faked dir is still in $PATH")
+
+    def test_which(self):
+        """test if which returns a path that corresponds to unix which"""
+
+        testnames = ["python", "java"]
+
+        for scriptname in testnames:
+            mpiwhich = mpim.which(scriptname)
+            exitcode, unixwhich = run_simple("which " + scriptname)
+            if exitcode > 0:
+                raise Exception("Something went wrong while trying to run `which`: %s" % unixwhich)
+
+            self.assertTrue(mpiwhich, msg="mpi which did not return anything, (unix which: %s" % unixwhich)
+            self.assertEqual(mpiwhich + "\n", unixwhich,
+                             msg="the return values of unix which and which() aren't"" the same: %s != %s" %
+                             (mpiwhich + "\n", unixwhich))
+
+     ###################
+     ## MPI functions ##
+     ###################
+
+    def test_options(self):
+        """running mympirun with bad options"""
+        optionparser = MympirunOption()
+        optionparser.args = ['echo', 'foo', 'zever']
+        # should not throw an error
+        try:
+            mpi_instance = getinstance(mpim.MPI, Local, optionparser)
+            mpi_instance.main()
+        except Exception:
+            self.fail("mympirun raised an exception while running main()")
+
+        optdict = mpi_instance.options.__dict__
+
+        print("args given to mympirunoption: %s, instance options: %s, " % (optionparser.args, optdict))
+
+        for opt in optionparser.args:
+            self.assertFalse(opt in optdict)
+
+    # def test_is_mpirun_for(self):
```

Comment:
I couldn't find a way to correctly load and unload the modules required for this test